app/csv_generator/pacbio_sample_sheet_message.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Creates a message in the correct structure for the warehouse
+class PacbioSampleSheetMessage < DataStructureBuilder
+  # return a list of column names ie headers
+  # eg ['System Name', 'Run Name']
+  def csv_headers
+    configuration.column_order
+  end
+
+  # Data Structure as csv
+  def payload
+    CSV.generate do |csv|
+      csv << csv_headers
+      # data_structure = {"sorted_wells"=>[{"Library Type"=>"Standard",
```

Comment:
This is recursive so maybe you can turn it into a recursive function? As long as you have a terminator it should be fine? It would also make Robocop happy.